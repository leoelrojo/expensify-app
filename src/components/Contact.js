import React from 'react';

const Contact = () => (
    <div>Please, leave me a message</div>
);

export default Contact;