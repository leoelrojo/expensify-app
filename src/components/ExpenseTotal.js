import React from 'React';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectSumExpenses from '../selectors/expenses-total';

const statementToReturn = (props) => {
    if(props.expenses.length === 0) {
        return "No expenses"
    } else {
        return (
            props.expenses.length > 1 
            ? `${props.expenses.length} expenses for ${
                numeral(props.expenses.sum / 100)
                .format('0.00$')
            }` 
            : `${props.expenses.length} expense for ${
                numeral(props.expenses.sum / 100)
                .format('0.00$')
            }`
        )
    }
};

export const ExpenseTotal = (props) => ( 
    <div>
        {
            statementToReturn(props)
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectSumExpenses(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpenseTotal);