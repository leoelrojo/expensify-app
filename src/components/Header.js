import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
<<<<<<< HEAD
        <h1>Expensify App</h1>
        <NavLink to="/" exact={true} activeClassName="is-active">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
=======
        <h1>Portfolio App</h1>
        <NavLink to="/" activeClassName='is-active' exact={true}>Home</NavLink>
        <NavLink to="/portfolio" activeClassName='is-active' exact={true}>Portfolio</NavLink>
        <NavLink to="/contact" activeClassName='is-active'>Contact</NavLink>
>>>>>>> d24604c3df50f3bbe23d64f1d743d82d44bb48ca
    </header>
);

export default Header;