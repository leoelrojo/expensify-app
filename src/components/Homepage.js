import React from 'react';

const Homepage = () => (
    <div>
        Welcome to my portfolio!
    </div>
);

export default Homepage;