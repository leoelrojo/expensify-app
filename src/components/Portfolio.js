import React from 'react';
import { Link } from 'react-router-dom';

const Portfolio = () => (
    <div>
        <div>Checkout the following things I've done:</div>
        <Link to="/portfolio/1">Stuff 1</Link>
        <Link to="/portfolio/2">Stuff 2</Link>
    </div>
);

export default Portfolio;