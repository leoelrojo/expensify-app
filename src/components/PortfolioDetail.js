import React from 'react';

const PortfolioDetail = (props) => (
    <div>
        Displaying portfolio {props.match.params.id}
    </div>
);

export default PortfolioDetail;