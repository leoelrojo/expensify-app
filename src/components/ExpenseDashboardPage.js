import React from 'react';
import ExpenseTotal from './ExpenseTotal';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseTotal />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;