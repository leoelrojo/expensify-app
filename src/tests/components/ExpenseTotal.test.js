import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseTotal } from '../../components/ExpenseTotal';
import expenses from '../fixtures/expenses';

test('should render statement with no expenses at all', () => {
    const wrapper = shallow(<ExpenseTotal expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render statement with only 1 expense (singular)', () => {
    const wrapper = shallow(<ExpenseTotal expenses={expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render statement with the sum of expenses (plural)', () => {
    const wrapper = shallow(<ExpenseTotal expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});