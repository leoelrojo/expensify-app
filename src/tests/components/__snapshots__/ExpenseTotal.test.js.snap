// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render statement with no expenses at all 1`] = `
<div>
  No expenses
</div>
`;

exports[`should render statement with only 1 expense (singular) 1`] = `
<div>
  undefined expense for 0.00$
</div>
`;

exports[`should render statement with the sum of expenses (plural) 1`] = `
<div>
  3 expenses for 0.00$
</div>
`;
